lib/jobs.py

```diff
@@ -0,0 +1,276 @@
+    for arg, value in kwargs.items():
+        switch = f"--{re.sub('_', '-', arg)}"
+        if value is None:
+            continue
+        if isinstance(value, bool):
+            if value:
+                cmd.append(switch)
+            continue
+
+        cmd.append(switch)
+        if isinstance(value, list):
+            logging.error(value)
+            cmd.extend(str(v) for v in value)
+        else:
+            cmd.append(str(value))
+    return cmd
+
+
+def fill_job(job):
+    parser = argparse.ArgumentParser()
+    for group_name, args in get_add_job_args():
+        group = parser.add_argument_group(title=group_name)
+        for arg in args:
+            flags = arg.pop("flags")
+            group.add_argument(*flags, **arg)
+    args = configure_args(**job)
+    job_dict = vars(parser.parse_known_args(args)[0])
+    for key in job:
+        if key not in job_dict:
+            job_dict[key] = job[key]
+    return job_dict
+
+
+async def add_job(job_dict):
+    cache_file = litani.get_cache_dir() / litani.CACHE_FILE
+    with open(cache_file) as handle:
+        cache_contents = json.load(handle)
+    if job_dict["ci_stage"] not in cache_contents["stages"]:
+        valid_stages = "', '".join(cache_contents["stages"])
+        logging.error(
+            "Invalid stage name '%s' was provided, possible "
+            "stage names are: '%s'", job_dict["ci_stage"], valid_stages)
+        sys.exit(1)
+
+    jobs_dir = litani.get_cache_dir() / litani.JOBS_DIR
+    jobs_dir.mkdir(exist_ok=True, parents=True)
+
+    if job_dict["phony_outputs"]:
+        if not job_dict["outputs"]:
+            job_dict["outputs"] = job_dict["phony_outputs"]
+        else:
+            for phony_output in job_dict["phony_outputs"]:
+                if phony_output not in job_dict["outputs"]:
+                    job_dict["outputs"].append(phony_output)
+
+    if "func" in job_dict:
+        job_dict.pop("func")
+
+    job_id = str(uuid.uuid4())
+    job_dict["job_id"] = job_id
+    job_dict["status_file"] = str(
+        litani.get_status_dir() / ("%s.json" % job_id))
+
+    logging.debug("Adding job: %s", json.dumps(job_dict, indent=2))
+
+    for key in _PRIVATE_JOB_FIELDS:
+        if key not in job_dict:
+            raise AssertionError(f"Key {key} missing from job definition")
+
+    with litani.atomic_write(jobs_dir / ("%s.json" % job_id)) as handle:
+        print(json.dumps(job_dict, indent=2), file=handle)
+
+
+async def get_jobs():
+    out = []
+    jobs_dir = litani.get_cache_dir() / litani.JOBS_DIR
+
+    if not jobs_dir.exists():
+        logging.warning("No jobs have been added")
+        return out
+
+    for job in jobs_dir.iterdir():
+        with open(job) as handle:
+            job_dict = json.load(handle)
+            for key in _PRIVATE_JOB_FIELDS:
+                job_dict.pop(key)
+            out.append(job_dict)
+    return out
+
+
+async def print_jobs(args=None):
+    jobs = await get_jobs()
+
+    out_file = sys.stdout
```

Comment:
I don't think you need this code? The argparser already takes care of it, you've set the default to be stdout and the type to be a file to write to

Reply:
I call `print_jobs()` from `transform_jobs()` as well, which doesn't pass in any arguments